batcher/aligned-sdk/src/sdk.rs

```diff
@@ -99,6 +111,103 @@ pub async fn submit_multiple_and_wait_verification(
     Ok(aligned_verification_data)
 }
 
+/*
+Returns the estimated `max_fee` depending on the batch inclusion preference of the user, based on the max priority gas price.
+NOTE: The `max_fee` is computed from an rpc nodes max priority gas price.
+
+For batch cost estimates we estimate a batch will have ~32 proofs present.
+The `max_fee` estimates therefore are:
+    - Min: Specifies a `max_fee` equivalent to the cost of 1 proof in a 32 proof batch.
+        This estimates the lowest possible cost the user should specify for there proof with lowest priority.
+    - Default: Specifies a `max_fee` equivalent to the cost of 10 proofs in a 32 proof batch.
+        This estimates the cost the user should specify for inclusion within the batch.
+    - Instant: specifies a `max_fee` equivalent to the cost of all proofs within in a 32 proof batch.
+        This estimates the cost the user should specify to pay for the entire batch of proofs and have there proof included instantly.
+*/
+pub async fn estimate_max_fee(
+    eth_rpc_url: &str,
+    estimate: PriceEstimate,
+) -> Result<U256, errors::SubmitError> {
+    // Price of 1 proof in 32 proof batch
+    let fee_per_proof = match fee_per_proof(eth_rpc_url, 32).await {
+        Ok(price) => price,
+        Err(e) => {
+            return Err(SubmitError::GenericError(format!(
+                "Failed to compute `fee_per_proof`: {:?}",
+                e
+            )))
+        }
+    };
+
+    let proof_price = match estimate {
+        PriceEstimate::Min => fee_per_proof,
+        PriceEstimate::Default => U256::from(10) * fee_per_proof,
+        PriceEstimate::Instant => U256::from(32) * fee_per_proof,
```

Comment:
I will guess: this is the batch size, because "instant" means "close the batch now, even if this is the only proof, I'll pay for the whole batch if necessary", right?

Reply:
some docs explaining these multipliers would be nice

Reply:
Extracted to constants